docs/operators_and_observables.ipynb

```diff
@@ -789,7 +803,7 @@
     "collector = cirq.PauliSumCollector(circuit, psum, samples_per_term=10_000)\n",
     "\n",
     "# Provide a sampler. See also: collector.collect(...).\n",
-    "collector.collect_async(sampler=cirq.Simulator())\n",
+    "collector.collect(sampler=cirq.Simulator())\n",
```

Comment:
I tried to run this cell, but it's taking an unreasonably long time to complete (5+ minutes). To minimize the impact on test times, we should reduce `samples_per_term` to a number that allows this to complete in 30 seconds or less.

If this requires choosing an unrealistically low value for `samples_per_term`, please also add a comment explaining why we're using the new value.

Reply:
That is strange. It is super quick to run that. In google colab that line takes less than 300ms. 
![image](https://user-images.githubusercontent.com/16711681/114925312-cb5def00-9de3-11eb-9e24-4def2d9ff5b2.png)
In the github action, the whole notebook execution is like one and half minutes.
![image](https://user-images.githubusercontent.com/16711681/114925378-e16baf80-9de3-11eb-8183-6433cd9bc3f6.png)

I wonder why it takes you so long time to complete. (I guess maybe related to live lock?) Could you show your test environment?

Reply:
I'm running this directly from the [Colab link](https://colab.sandbox.google.com/github/quantumlib/Cirq/blob/c6cbbfc73fc92297a5708926ffb42795d73a0b5e/docs/operators_and_observables.ipynb) (modified to point to the latest commit on this PR) and the execution consistently hangs or crashes - I haven't had a run complete successfully yet.

Logs include `IndexError: pop from an empty deque`, but it's unclear whether that's coming from the Colab restart or from asyncio.

Reply:
For clarity, I've been running the linked colab with the `Runtime > Run all` command.

Reply:
Interesting. Yes, if I click `Run all`, it will fail. But if click `Run before` that Cell and then manually `shift+Enter` run that cell can be finished quickly. I guess `Run all` added another layer of running environment caused this. I will take a look to understand this better.

Reply:
> Interesting. Yes, if I click `Run all`, it will fail. But if click `Run before` that Cell and then manually `shift+Enter` run that cell can be finished quickly.

I can reproduce this result - `Run before` and then running this cell also succeeds for me. Thanks for digging into this!

Reply:
I still do not understand why this happens. What I found is the [TensorFlow Federated colab](https://www.tensorflow.org/federated/tutorials/federated_learning_for_image_classification#before_we_start) has encountered similar asynchrony issue and they used nest_asyncio as well for solution. I tried to use `Run all` their colab, and they do not have similar issue.